sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/_operations/_patch.py

```diff
@@ -0,0 +1,270 @@
+    you can't accomplish using the techniques described in
+    https://aka.ms/azsdk/python/dpcodegen/python/customize
+    """
+
+
+class BaseStatePollingMethod:
+    """Base polling method for methods returning responses containing a 'state' field; the polling
+    completes when 'state' becomes a desired value.
+    """
+
+    def __init__(
+        self,
+        operation: Callable[[], Any],
+        desired_state: str,
+        polling_interval_s: float,
+    ):
+        self._operation = operation
+        self._desired_state = desired_state
+        self._polling_interval_s = polling_interval_s
+
+        self._deserialization_callback = None
+        self._status = "constructed"
+        self._latest_response = {}
+
+    def initialize(self, client, initial_response, deserialization_callback):  # pylint: disable=unused-argument
+        self._evaluate_response(initial_response)
+        self._deserialization_callback = deserialization_callback
+
+    def _evaluate_response(self, response: JSON) -> None:
+        self._status = "finished" if response["state"] == self._desired_state else "polling"
+        self._latest_response = response
+
+    def status(self) -> str:
+        return self._status
+
+    def finished(self) -> bool:
+        return self.status() in {"finished", "failed"}
+
+    def resource(self):
+        if self._deserialization_callback:
+            return self._deserialization_callback(self._latest_response)
+
+        return self._latest_response
+
+
+class StatePollingMethod(BaseStatePollingMethod, PollingMethod):
+    """Polling method for methods returning responses containing a 'state' field; the polling
+    completes when 'state' becomes a desired value.
+    """
+
+    def __init__(
+        self,
+        operation: Callable[[], JSON],
+        desired_state: str,
+        polling_interval_s: float,
+    ):
+        super().__init__(operation, desired_state, polling_interval_s)
+
+    def run(self) -> None:
+        try:
+            while not self.finished():
+                response = self._operation()
+                self._evaluate_response(response)
+
+                if not self.finished():
+                    time.sleep(self._polling_interval_s)
+        except Exception:
+            self._status = "failed"
+            raise
+
+
+class ConfidentialLedgerClientOperationsMixin(GeneratedOperationsMixin):
+    def begin_get_ledger_entry(
+        self, transaction_id: str, *, collection_id: Optional[str] = None, **kwargs: Any
+    ) -> LROPoller[JSON]:
+        polling = kwargs.pop("polling", True)  # type: Union[bool, PollingMethod]
+        lro_delay = kwargs.pop("polling_interval", 0.5)
+
+        def operation() -> JSON:
+            return super(ConfidentialLedgerClientOperationsMixin, self).get_ledger_entry(
+                transaction_id, collection_id=collection_id, **kwargs
+            )
+
+        initial_response = operation()
```

Comment:
you can just do `initial_response = super().get_ledger_entry(...)`

Reply:
I wanted to call `operation` so I only have to write the method call once + explicitly make sure the calls are exactly the same.